File: porting-console-app-rhill34/visibility/HourlyEmployee.cs
using System;
namespace visibility
{
    public class HourlyEmployee : Employee
    {
        public HourlyEmployee(double hours, double wages, string fName,
            string lName, Address address, Position position) :
            base(fName, lName, address, position)
        {
            Hours = hours;
            Wage = wages;
        }

        //Properties
        public double Hours { get; set; }
        public double Wage { get; set; }

        public double CalculatePay()
        {
            double total = Hours * Wage;

            if(Hours > 40)
            {
                total += 0.5 + (Hours - 40);
                return total;
            }

            return total;
        }

        public override string ToString()
        {
            return base.ToString() + "Hours :" + Hours + Environment.NewLine +
                "Paid : " + Wage + Environment.NewLine +
                "Expected Pay is " + CalculatePay();
        }
    }
}

File: porting-console-app-rhill34/javaFiles/visibility/HourlyEmployee.java
package visibility;

public class HourlyEmployee extends Employee {

    private double hours;
    private double wages;

    public HourlyEmployee(String fName, String lName, Address address, Position position, double hours, double wages) {
        super(fName, lName, address, position);
        this.hours=hours;
        this.wages=wages;

    }

    public double calculatePay() {
        double total = hours * wages;

        if(hours > 40)
            total += 0.5 * (hours - 40);

        return total;
    }

    public String toString() {
        String lname = super.lName;
        String fname = super.fName;
        return "Hourly"+super.toString();
    }

}
